### DISCUSSION TITLE
Is there a list of variables for template syntax?

### DISCUSSION THREAD

**Discussion by AvihaiSam** [vectordotdev/vector#18113] [Q&A] [answered]
there're some examples over the resources documentation like `application_id` and `host`. is there a complete list somewhere? couldn't find any...

**Comment by jszwedko** [comment, marked as answer]
Hi @AvihaiSam ,

The available fields are any that exist on the event. This can vary depending on the source of the data and any transformations that have been applied. The easiest way to see which fields are available are to use `vector tap` to see the events flowing into a component (or wire up the `console` sink with `encoding.codec = "json"`).

**Reply by AvihaiSam** [reply]
@jszwedko ok. If I understand correctly - I can:
1. access all the fields that are generated by the current component using their name without any prefix
2. access any other (previously generated) event fields with `.` prefix
3. access any metadata fields with `%` prefix.
Am I correct?

BTW, is there a way to show the metadata fields on `vector tap`?
I tried `vector tap -m` but it doesn't show the metadata fields which can be accessed with `%` prefix.

**Reply by jszwedko** [reply]
That's mostly correct. 1 and 2 are actually the same though. The `.` prefix is optional for backwards compatibility reasons, but is recommended.

Unfortunately `vector tap` does not show metadata yet. That is something we'd like to add.
